src/Symfony/Component/Console/Formatter/OutputFormatter.php

```diff
@@ -231,24 +232,45 @@ private function createStyleFromString(string $string)
     /**
      * Applies current style from stack to text, if must be applied.
      */
-    private function applyCurrentStyle(string $text, string $current, int $width): string
+    private function applyCurrentStyle(string $text, string $current, int $width, int &$length): string
     {
         if ('' === $text) {
             return '';
         }
 
-        if ($width) {
-            if ('' !== $current) {
-                $text = ltrim($text);
-            }
+        if (!$width) {
+            return $this->isDecorated() ? $this->styleStack->getCurrent()->apply($text) : $text;
+        }
+
+        if ($length) {
+            $prefix = substr($text, 0, $i = $width - $length)."\n";
+            $text = substr($text, $i);
+        } else {
+            $prefix = '';
+        }
 
-            $text = wordwrap($text, $width, "\n", true);
+        if ('' !== $current) {
+            $text = ltrim($text);
+        }
 
-            if ('' !== $current && "\n" !== substr($current, -1)) {
-                $text = "\n".$text;
-            }
+        $text = $prefix.substr(preg_replace('~(.{'.$width.'})\s*~', "\$1\n", $text."\n"), 0, -1);
+
+        if (!$length && '' !== $current && "\n" !== substr($current, -1)) {
+            $text = "\n".$text;
+        }
+
+        if (!$this->isDecorated()) {
+            return $text;
+        }
+
+        $lines = array();
+        foreach (explode("\n", $text) as $line) {
+            $lines[] = '' === $line ? $line : $this->styleStack->getCurrent()->apply($line);
```

Comment:
here we fix #27832 by formatting each line separate. This only helps when the user wraps a strings. If it's wrapped by the terminal it can still occur.